presto-verifier/src/main/java/com/facebook/presto/verifier/prestoaction/JdbcPrestoAction.java

```diff
@@ -144,10 +144,8 @@ private PrestoConnection getConnection(QueryStage queryStage)
             // Do nothing
         }
 
-        Map<String, String> sessionProperties = ImmutableMap.<String, String>builder()
-                .putAll(queryConfiguration.getSessionProperties())
-                .put(QUERY_MAX_EXECUTION_TIME, getTimeout(queryStage).toString())
-                .build();
+        Map<String, String> sessionProperties = new HashMap<>(queryConfiguration.getSessionProperties());
+        sessionProperties.put(QUERY_MAX_EXECUTION_TIME, getTimeout(queryStage).toString());
```

Comment:
@caithagoras It is possible that a query would run with a timeout that's a lot larger than `getTimeout(queryStage)`. Is this OK? Why? Perhaps, add a comment here.

Reply:
Added comment.
```
Add or override query max execution time to enforce the timeout.
```

From this `PrestoAction` class's perspective, it only cares about its contract, that is, run a given query or fail if the query execution time exceeds the given timeout.

From the verification perspective, a long-running query will fail on the control, which will be skipped, excluded from our consideration.